src/dialects/postgres/query-generator-typescript.ts

```diff
@@ -48,4 +51,38 @@ export class PostgresQueryGeneratorTypeScript extends AbstractQueryGenerator {
       'GROUP BY i.relname, ix.indexrelid, ix.indisprimary, ix.indisunique, ix.indkey ORDER BY i.relname;',
     ]);
   }
+
+  removeIndexQuery(tableName: TableNameOrModel, indexNameOrAttributes: string | string[], options: RemoveIndexQueryOptions) {
+    if (options) {
+      rejectInvalidOptions(
+        'removeIndexQuery',
+        this.dialect.name,
+        REMOVE_INDEX_QUERY_SUPPORTABLE_OPTIONS,
+        new Set(['concurrently', 'ifExists', 'cascade']),
+        options,
+      );
+
+      if (options.cascade && options.concurrently) {
+        throw new Error(`Cannot specify both concurrently and cascade options in removeIndexQuery for ${this.dialect.name} dialect`);
+      }
+    }
+
+    let indexName;
+    const table = this.extractTableDetails(tableName);
+    if (Array.isArray(indexNameOrAttributes)) {
+      indexName = generateIndexName(table, { fields: indexNameOrAttributes });
+    } else {
+      indexName = indexNameOrAttributes;
+    }
+
+    return joinSQLFragments([
+      'DROP INDEX',
+      options?.concurrently ? 'CONCURRENTLY' : '',
+      options?.ifExists ? 'IF EXISTS' : '',
+      table?.schema && table.schema !== this.dialect.getDefaultSchema()
+        ? `${this.quoteIdentifier(table.schema)}.${this.quoteIdentifier(indexName)}`
+        : this.quoteIdentifier(indexName),
```

Comment:
```suggestion
      `${this.quoteIdentifier(table.schema)}.${this.quoteIdentifier(indexName)}`
```

@ephys I think we discussed that we want the schema specified as much as possible, right?

Reply:
Yes :) we don't want ambiguity caused by searchPath being set to something else than the default

Reply:
`table.schema` type is `string | undefined` so do you want me to use the schema if defined otherwise the default schema or ignore the schema if its not definied.
Something like this:
```ts
table?.schema
  ? `${this.quoteIdentifier(table.schema)}.${this.quoteIdentifier(indexName)}`
  : this.quoteIdentifier(indexName),
````
or
```ts
`${this.quoteIdentifier(table?.schema || this.dialect.getDefaultSchema())}.${this.quoteIdentifier(indexName)}`,
```

Reply:
Ah, I think we might need to update the return type of `extractTableDetails` so TypeScript knows that we always return a schema (since it does). For now `table.schema!` should work

Reply:
Updated now